53 organisation of folders

SuperCreativeUniqueUsername opened the pull request:
i think i was doing too much on this branch, this was initially just for file organisation but turned into trying to get the nav working between the two permissions (user and organiser). the foundation is done for this in the future but i think for now its would be more important to merge this so that we can all start working off the new file structure.

SuperCreativeUniqueUsername commented on app/src/main/java/com/example/pygmyhippo/MainActivity.java:
```diff
@@ -10,27 +16,93 @@
-        BottomNavigationView navView = findViewById(R.id.nav_view);
-        // Passing each menu ID as a set of Ids because each
-        // menu should be considered as top level destinations.
+    @Override
```
This is the problem code that would need to be reviewed in the future

katzhong commented on app/build.gradle.kts:
```diff
@@ -6,6 +6,10 @@ android {
     namespace = "com.example.pygmyhippo"
     compileSdk = 34
 
+    dataBinding {
```
curious, what does this do?

SuperCreativeUniqueUsername replied:
I think this may have been added automatically, I am not sure becuase I don't think I wrote that. https://stackoverflow.com/q/59390492 apparently it's old, I'll look into seeing what happens if this is changed to the new way.

SuperCreativeUniqueUsername commented:
UML diagram is updated with main classes, CRC is updated, storyboard is not. Classes are commented. 
![image](https://github.com/user-attachments/assets/491a9b85-e766-45d6-99ef-e6cc46443e00)

SuperCreativeUniqueUsername commented:
UML: https://drive.google.com/file/d/11mWBm0JmifDqCy7MJHijSQ3drCAQO-Qm/view?usp=sharing

Kori-Kozicki reviewed (approved):
I think this looks good. It's alot to check over but I agree this is a good base to have for the rest of our code and a good start to navigation

katzhong commented on app/src/main/java/com/example/pygmyhippo/MainActivity.java:
```diff
@@ -10,27 +16,107 @@
+public class MainActivity extends AppCompatActivity implements OnRoleSelectedListener {
+
+    private OrganiserMainActivityNagivationBinding organiserBinder;
+    private UserMainActivityNagivationBinding userBinding;
```
NIT: any reason why it's orgniserBindER and userBindING? if not, perhaps having consistent naming to what these are would be good

SuperCreativeUniqueUsername replied:
you're right

katzhong commented on app/src/main/java/com/example/pygmyhippo/user/MyEventsFragment.java:
```diff
@@ -9,7 +9,7 @@
 import androidx.fragment.app.Fragment;
 
-import com.example.pygmyhippo.databinding.FragmentMyeventsBinding;
+import com.example.pygmyhippo.databinding.UserFragmentMyeventsBinding;
```
NIT: naming should be UserFragmentMyEventsBinding?

SuperCreativeUniqueUsername replied:
yes because there are two fragments my event binding, one for organiser and one for user

katzhong replied:
oops should specify, did you mean for events to be capitalized or no

SuperCreativeUniqueUsername replied:
this is auto generated from the xml file user_fragment_myevents.xml in the layout folder. are you suggesting i change the name of that xml file?

katzhong commented on app/src/main/res/layout/user_fragment_myevents.xml:
```diff
@@ -0,0 +1,22 @@
+    tools:context=".user.MyEventsFragment">
+
+    <TextView
+        android:id="@+id/text_home"
```
NIT: should the naming of this be something more specific so that it is more identifiable?

SuperCreativeUniqueUsername replied:
this was added missed from jens pr, since this was just a copy paste of her myevents fragment.

katzhong commented on app/src/main/res/layout/user_fragment_qr.xml:
```diff
@@ -0,0 +1,22 @@
+    tools:context=".user.QRFragment">
+
+    <TextView
+        android:id="@+id/text_dashboard"
```
NIT: same here?

katzhong commented on app/src/main/res/menu/organiser_bottom_nav_menu.xml:
```diff
@@ -0,0 +1,19 @@
+        android:title="@null"/>
+
+    <item
+        android:id="@+id/navigation_dashboard"
```
just wondering, why was the name dashboard chosen for add event and home for the calendar?

SuperCreativeUniqueUsername replied:
this was from jens pr too, i guess I missed this when fixing things up

SuperCreativeUniqueUsername commented:
I changed the organiser side of naming, I added fixmes on the user side of naming. I don't want to change names for user since that is not what I'm working on and I'm worried renaming could mess up stuff for whoever is working on the user side

katzhong reviewed (approved):
assuming this works...
